core/metacling/src/TClingCallFunc.cxx

```diff
@@ -1195,23 +1209,25 @@ tcling_callfunc_ctor_Wrapper_t TClingCallFunc::make_ctor_wrapper(const TClingCla
    //          *ret = new ClassName[nary];
    //       }
    //    }
-   ++indent_level;
    indent(buf, indent_level);
    buf << "if (!arena) {\n";
    ++indent_level;
    indent(buf, indent_level);
    buf << "if (!nary) {\n";
    ++indent_level;
    indent(buf, indent_level);
-   buf << "*ret = new " << class_name << ";\n";
+   buf << "*ret = new " << class_name << constr_arg << ";\n";
    --indent_level;
    indent(buf, indent_level);
    buf << "}\n";
    indent(buf, indent_level);
    buf << "else {\n";
    ++indent_level;
    indent(buf, indent_level);
-   buf << "*ret = new " << class_name << "[nary];\n";
+   if (constr_arg.empty())
+      buf << "*ret = new " << class_name << "[nary];\n";
+   else
+      buf << "*ret = nullptr;\n"; // no way to create objects array with non-default constructor
```

Comment:
You would need to allocate via new char[nary*sizeof(...)] and then iterate over calls using operator new with placement. (see for example ore Effective C++ - Scott Meyers: Item 4 - Avoid gratuitous default constructors)

Reply:
Thanks, not I provide a solution for such classes